nemo_rl/models/policy/dtensor_policy_worker.py

```diff
@@ -1856,6 +1856,7 @@ def prepare_for_training(self, *args, **kwargs) -> None:
             hasattr(self, "optimizer")
             and self.optimizer is not None
             and not self.cpu_offload
+            and self.cfg.get("offload_optimizer_states_for_logprob", False)
```

Comment:
```suggestion
            and not self.cfg["offload_optimizer_states_for_logprob"]
```

Thanks for the update! I also prefer set it in config than through env var.

And I suggest removing the default value in the code, so that if people somehow forget to set the param when it will cause OOM, they will get an error said missing this offload param to guide them set it instead of just getting OOM and don't know it's implicitly set in the code.